pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -232,17 +236,28 @@ func (c *NetworkConfig) RunProxy() {
 }
 
 // getNodeIP is copied from the upstream proxy config to retrieve the IP of a node.
-func getNodeIP(client kv1core.CoreV1Interface, hostname string) net.IP {
-	var nodeIP net.IP
-	node, err := client.Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+func getNodeIP(client kv1core.CoreV1Interface, hostname string) (net.IP, error) {
+	var node *v1.Node
+	var nodeErr error
+
+	// This will wait 2, 4, 8, ... 256 seconds, for a total of 8.5 mins
+	nodeWaitBackoff := utilwait.Backoff{
+		Duration: 2 * time.Second,
+		Factor:   2,
+		Steps:    8,
 	}
-	nodeIP, err = utilnode.GetNodeHostIP(node)
+	utilwait.ExponentialBackoff(nodeWaitBackoff, func() (bool, error) {
+		node, nodeErr = client.Nodes().Get(hostname, metav1.GetOptions{})
+		return nodeErr == nil, nil
+	})
+	if nodeErr != nil {
+		return nil, fmt.Errorf("Failed to retrieve node info: %v", nodeErr)
+	}
+
+	nodeIP, err := utilnode.GetNodeHostIP(node)
```

Comment:
Do we actually fail to 'Get' the node itself, or do we fail to get its HostIP?

Reply:
I presume its only the 'Get' which is racing and we do not have to wait for HostIP fetching (status etc).

Reply:
Yeah, it's only the Get that is the problem.  We are reading the IP address that is reported at node creation so it's there when the node is there.